pkg/secretsmanager/secretsmanager_integration_test.go

```diff
@@ -148,3 +156,76 @@ func TestEnsureGCPSecrets(t *testing.T) {
 		t.Fatalf("Expected 'admin', got: '%s'", data)
 	}
 }
+
+// newAzureClient create an Azure client with an authorizer from the environment
+func newCliAzureClient() (*keyvault.BaseClient, error) {
+	authorizer, err := azauth.NewAuthorizerFromCLI()
+	if err != nil {
+		return &keyvault.BaseClient{}, err
+	}
+	client := keyvault.New()
+	client.Authorizer = authorizer
+	return &client, nil
+}
+
+// TestLoadAzure requires the vault to be pre-configured at the moment and for hard deletes too
+// az keyvault create --name "secret-agent-test" --resource-group "secret-agent-test" --location eastus --enable-soft-delete false
+func TestLoadAzure(t *testing.T) {
+	keyConfig := &types.KeyConfig{
+		Name: "username",
+	}
+	secretConfig := &types.SecretConfig{
+		Name:      "load-azure-secrets",
+		Namespace: "default",
+		Keys:      []*types.KeyConfig{keyConfig},
+	}
+	node := &types.Node{
+		Path:         []string{"load-azure-secrets", "username"},
+		SecretConfig: secretConfig,
+	}
+	nodes := []*types.Node{node}
+
+	vaultName := "secret-agent-test"
+	// expect nothing found
+	baseCtx := context.Background()
+	ctx1, cancel := context.WithTimeout(baseCtx, 3000000000)
+	defer cancel()
+	client, err := newCliAzureClient()
+	err = loadAzureSecrets(ctx1, client, vaultName, nodes)
+	if err != nil {
+		t.Fatalf("Expected no error, got: %+v", err)
+	}
+	if len(node.Value) != 0 {
+		t.Fatalf("Expected no value, got: %+v", node.Value)
+	}
+
+	// create a secret to load
+	var secParams keyvault.SecretSetParameters
```

Comment:
If you want to avoid the mixed variable instantiation syntax here, you can `secParams := &keyvault.SecretSetParameters{}`.

Reply:
Personally I like to use the `myvar := blah` syntax inside functions, and the `var myvar blah` syntax outside functions (since it's required outside functions.)

Reply:
me too; copy pasta...